src/stc/PlatWX.cpp

```diff
@@ -1983,433 +1997,743 @@ PRectangle Window::GetMonitorRect(Point pt) {
+
+    wxBitmap bmp(img);
+    RegisterImage(type, bmp);
+}
+
+void wxSTCListBoxVisualData::RegisterRGBAImage(int type, int width, int height,
+                                    const unsigned char *pixelsImage)
+{
+    wxBitmap bmp = BitmapFromRGBAImage(width, height, pixelsImage);
+    RegisterImage(type, bmp);
+}
+
+void wxSTCListBoxVisualData::ClearRegisteredImages()
+{
+    m_imgList.clear();
+}
+
+const wxBitmap* wxSTCListBoxVisualData::GetImage(int i) const
+{
+    ImgList::const_iterator it = m_imgList.find(i);
+
+    if ( it != m_imgList.end() )
+        return &(it->second);
+    else
+        return NULL;
+}
+
+void wxSTCListBoxVisualData::ComputeColours()
+{
+    // wxSYS_COLOUR_BTNSHADOW seems to be the closest match with most themes.
+    m_borderColour = wxSystemSettings::GetColour(wxSYS_COLOUR_BTNSHADOW );
+
+    if ( !m_bgColourIsSet )
+        m_bgColour = wxSystemSettings::GetColour(wxSYS_COLOUR_LISTBOX);
+
+    if ( !m_textColourIsSet )
+        m_textColour = wxSystemSettings::GetColour(wxSYS_COLOUR_LISTBOXTEXT);
 
-    virtual bool Show(bool show = true)
+    if ( m_hasListCtrlAppearance )
     {
-        bool rv = wxFrame::Show(show);
-        if (rv && show)
-            ActivateParent();
-#ifdef __WXMAC__
-        GetParent()->Refresh(false);
-#endif
-        return rv;
+        // If m_highlightBgColour and/or m_currentBgColour are not
+        // explicitly set, set them to wxNullColour to indicate that they
+        // should be drawn with wxRendererNative.
+        if ( !m_highlightBgColourIsSet )
+            m_highlightBgColour = wxNullColour;
+
+        if ( !m_currentBgColourIsSet )
+            m_currentBgColour = wxNullColour;
+
+        #ifdef __WXMSW__
+            if ( !m_highlightTextColourIsSet )
+                m_highlightTextColour =
+                    wxSystemSettings::GetColour(wxSYS_COLOUR_LISTBOXTEXT);
+        #else
+            if ( !m_highlightTextColourIsSet )
+                m_highlightTextColour = wxSystemSettings::GetColour(
+                    wxSYS_COLOUR_LISTBOXHIGHLIGHTTEXT);
+        #endif
+
+        if ( !m_currentTextColour.IsOk() )
+            m_currentTextColour = wxSystemSettings::GetColour(
+                wxSYS_COLOUR_LISTBOXTEXT);
     }
-
-    void OnActivate(wxListEvent& WXUNUSED(event))
+    else
     {
-        doubleClickAction(doubleClickActionData);
+    #ifdef __WXOSX_COCOA__
+        if ( !m_highlightBgColourIsSet )
+            m_highlightBgColour = GetListHighlightColour();
+    #else
+        if ( !m_highlightBgColourIsSet )
+            m_highlightBgColour =
+            wxSystemSettings::GetColour(wxSYS_COLOUR_HIGHLIGHT);
+    #endif
+
+        if ( !m_highlightTextColourIsSet )
+            m_highlightTextColour =
+                wxSystemSettings::GetColour(wxSYS_COLOUR_LISTBOXHIGHLIGHTTEXT);
     }
+}
 
-    wxListView* GetLB() { return lv; }
+void SetColourHelper(bool& isSet, wxColour& itemCol, const wxColour& newColour)
```

Comment:
```suggestion
static void SetColourHelper(bool& isSet, wxColour& itemCol, const wxColour& newColour)
```
(or put it an unnamed namespace)